src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -396,12 +386,14 @@ public void MarkFieldValid(string key)
                 throw new ArgumentNullException(nameof(key));
             }
 
-            var modelState = GetModelStateForKey(key);
+            var modelState = GetModelStateForKey(key, createIfNotExists: true);
```

Comment:
I would have two different overloads, GetModelStateForKey and GetOrAddModelStateForKey, that way you avoid the boolean parameter and is less verbose.